### ISSUE TITLE
toLowerCase error when selecting a labeled contact/address to send ETH to

### ISSUE THREAD

**Issue by deluca-mike** [MetaMask/metamask-extension#9055]
**Describe the bug**
Can't send to a contact more than once if you name the recent address after the fist send to it

**To Reproduce (REQUIRED)**
1. Send ETH to an address
2. Label the recent address as a contact
3. Try to select or type that address as the recipient of a new ETH send

**Expected behavior**
A clear description of what you expected to happen.

**Screenshots**
None

**Browser details (please complete the following information):**
 - OS: OSX
 - Browser: Google Chrome Version 84.0.4147.89 (Official Build) (64-bit)
 - MetaMask Version: 8.0.4

**Additional context (Error Messages, etc.)**
```
Message: Cannot read property 'toLowerCase' of undefined

Code: TypeError

Stack:
TypeError: Cannot read property 'toLowerCase' of undefined
    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:1:3554870
    at Array.find (<anonymous>)
    at Function.mapToProps (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:1:3554848)
    at r (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1132363)
    at Function.r.mapToProps (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1132481)
    at r (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1132345)
    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1131189
    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1124589
    at Object.useMemo (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1002065)
    at useMemo (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1267371)
    at S (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1124533)
    at ha (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:997898)
    at Qa (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1006114)
    at Ka (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1005933)
    at Xa (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1005652)
    at ks (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1046575)
    at Mu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1028708)
    at ju (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1028633)
    at xu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1025973)
    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:984303
    at n.unstable_runWithPriority (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1283550)
    at Xi (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:984012)
    at Ji (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:984248)
    at Qi (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:984183)
    at se (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:1053235)
    at Cn (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:1:967834)
    at HTMLDocument.u (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:1:39
```

**Comment by deluca-mike** [comment]
Even after deleting the account, typing in the address as a recipient for an ETH send immediately causes the same error (on paste).

**Comment by brad-decker** [comment]
@deluca-mike thanks for reporting this bug! This will be resolved by #9062 v8.0.6. Related to #9019 in that fixing the underlying error resolves both bugs.
